apps/cowswap-frontend/src/legacy/state/price/updater.ts

```diff
@@ -79,25 +92,43 @@ function quoteUsingSameParameters(currentParams: FeeQuoteParams, quoteInfo: Quot
     hasSameReceiver
   // 2 checks: if there's a quoteInfo user address (meaning quote was already calculated once) and one without
   // in case user is not connected
-  return userAddress ? currentUserAddress === userAddress && paramsWithoutAddress : paramsWithoutAddress
+
+  const sameParams = userAddress ? currentUserAddress === userAddress && paramsWithoutAddress : paramsWithoutAddress
+
+  return {
+    fastQuoteSameParams: sameParams,
+    bestQuoteSameParams: sameParams && priceQuality === currentPriceQuality,
+  }
 }
 
+const FETCH_QUOTES = { fetchBestQuote: true, fetchFastQuote: true }
+const DONT_FETCH_QUOTES = { fetchBestQuote: false, fetchFastQuote: false }
+const FETCH_BEST_QUOTE = { fetchBestQuote: true, fetchFastQuote: false }
 /**
  *  Decides if we need to refetch the fee information given the current parameters (selected by the user), and the current feeInfo (in the state)
  */
 function isRefetchQuoteRequired(
   isLoading: boolean,
-  currentParams: FeeQuoteParams,
+  bestQuoteSameParams: boolean,
+  fastQuoteSameParams: boolean,
   quoteInformation?: QuoteInformationObject
-): boolean {
+): {
+  fetchBestQuote: boolean
+  fetchFastQuote: boolean
+} {
```

Comment:
I moved to this method the decision to fetch the best quote, the fast quote, or both